Examples/Autoregressive/ising1d_autoreg_conv.py

```diff
@@ -40,7 +40,7 @@
 # Variational state
 # With direct sampling, we don't need many samples in each step to form a
 # Markov chain, and we don't need to discard samples
-vs = nk.vqs.MCState(sa, ma, n_samples=64, n_discard=0)
+vs = nk.vqs.MCState(sa, ma, n_samples=64, n_discard_per_chain=0)
```

Comment:
I think that we should make sure that samplers have a field ```is_exact``` or something, telling us if the sampling is exact or approximate. If it is exact, ```MCState``` should set by default ```n_discard_per_chain=0```

Reply:
There's already some logic to handle this inside of [variational state](https://github.com/netket/netket/blob/6b1154873cd22baffd02e64e5e2105f665ee82e5/netket/vqs/mc_state.py#L318). We should just generalise it to handle not only exactsampler but also other samplers, indeed.

I'll do it in another PR though

Reply:
ok great, I would indeed not just check the class name but rather check the property of the sampler ```is_exact```, to be more general! yes another PR seems more appropriate

Reply:
@wdphy16 maybe you can do it? Add a property to `AbstractSampler` called `is_exact` defaulting to False. Override it for Autoregressive sampler and exact sampler to be True, and use it instead of an `isinstance` check to hard-set the n_discard to zero.

Reply:
Ok, let me do it